src/pkg/util/env/clean.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package env
+
+import (
+	"strings"
+
+	"github.com/opencontainers/runtime-tools/generate"
+	"github.com/singularityware/singularity/src/pkg/sylog"
+)
+
+// CleanEnv cleans environment variables before running the container
+func CleanEnv(g *generate.Generator, NoHome bool, IsCleanEnv bool, HomeDest string, environment []string) {
```

Comment:
I think this func should be called something like `SetContainerEnv()` instead of `CleanEnv()` to avoid confusion with the `--cleanenv` flag